File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int size = prices.size();
        int maxProfit  = 0;

        int minTillNow = prices[0];

        for(int i=1;i<size;i++)
        {
            minTillNow = min(minTillNow, prices[i]);

            int profit = prices[i]- minTillNow;
            maxProfit = max(maxProfit, profit);
        }
        return maxProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int last = prices[0];
        int profit = 0;
        int n = prices.length;

        for(int i=1;i<n;i++)
        {
            last = Math.min(last, prices[i]);
            profit = Math.max(prices[i] - last,profit);
        }
        return profit;
        
    }
}